matchms/SpectrumProcessor.py

```diff
@@ -0,0 +1,316 @@
+               msfilters.add_retention_time,
+               msfilters.derive_ionmode,
+               msfilters.correct_charge,
+               # msfilters.derive_adduct_from_name,  # run again? Or improve those filters?
+               # msfilters.derive_formula_from_name,  # run again? Or improve those filters?
+               msfilters.require_precursor_mz,
+               msfilters.add_parent_mass,
+               msfilters.harmonize_undefined_inchikey,
+               msfilters.harmonize_undefined_inchi,
+               msfilters.harmonize_undefined_smiles,
+               msfilters.repair_inchi_inchikey_smiles,
+               msfilters.repair_parent_mass_match_smiles_wrapper,
+               msfilters.require_correct_ionmode,
+               msfilters.require_precursor_below_mz,
+               msfilters.require_parent_mass_match_smiles,
+               msfilters.require_valid_annotation,
+               msfilters.normalize_intensities,
+               msfilters.select_by_intensity,
+               msfilters.select_by_mz,
+               msfilters.select_by_relative_intensity,
+               msfilters.remove_peaks_around_precursor_mz,
+               msfilters.remove_peaks_outside_top_k,
+               msfilters.require_minimum_number_of_peaks,
+               msfilters.require_minimum_of_high_peaks,
+               msfilters.add_fingerprint,
+               msfilters.add_losses,
+              ]
+
+FILTER_FUNCTIONS = {x.__name__: x for x in ALL_FILTERS}
+
+MINIMAL_FILTERS = ["make_charge_int",
+                   "interpret_pepmass",
+                   "derive_ionmode",
+                   "correct_charge",
+                   ]
+BASIC_FILTERS = MINIMAL_FILTERS \
+    + ["add_compound_name",
+       "derive_adduct_from_name",
+       "derive_formula_from_name",
+       "clean_compound_name",
+       "add_precursor_mz",
+    ]
+DEFAULT_FILTERS = BASIC_FILTERS \
+    + ["require_precursor_mz",
+       "add_parent_mass",
+       "harmonize_undefined_inchikey",
+       "harmonize_undefined_inchi",
+       "harmonize_undefined_smiles",
+       "repair_inchi_inchikey_smiles",
+       "repair_parent_mass_match_smiles_wrapper",
+       "normalize_intensities",
+    ]
+FULLY_ANNOTATED_PROCESSING = DEFAULT_FILTERS \
+    + [("require_correct_ionmode", {"ion_mode_to_keep": "both"}),
+       "require_parent_mass_match_smiles",
+       "require_valid_annotation",
+    ]
+
+PREDEFINED_PIPELINES = {
+    "minimal": MINIMAL_FILTERS,
+    "basic": BASIC_FILTERS,
+    "default": DEFAULT_FILTERS,
+    "fully_annotated": FULLY_ANNOTATED_PROCESSING,
+}
+
+
+class ProcessingReport:
+    """Class to keep track of spectrum changes during filtering.
+    """
+    def __init__(self):
+        self.counter_changed_field = defaultdict(int)
+        self.counter_added_field = defaultdict(int)
+        self.counter_removed_spectrums = defaultdict(int)
+        self.counter_number_processed = 0
+
+    def add_to_report(self, spectrum_old, spectrum_new: Spectrum,
```

Comment:
@florian-huber Filter steps that do not do any filtering do not end up in the report currently. This might be wanted behaviour, since it keeps the list short, however, you could also argue that it might be nice to have a complete overview of all the filter steps, to easily pick out the filter steps that do not have any effect.

Reply:
I created an issue about this #485